cmake/Dependencies.cmake

```
@@ -1024,11 +1024,18 @@ if(BUILD_PYTHON)
   endif()
 
   if(PYTHONINTERP_FOUND AND PYTHONLIBS_FOUND)
-    include_directories(SYSTEM ${PYTHON_INCLUDE_DIR})
+    add_library(python::python INTERFACE IMPORTED)
+    set_property(TARGET python::python PROPERTY
+        INTERFACE_INCLUDE_DIRECTORIES ${PYTHON_INCLUDE_DIRS})
+    set_property(TARGET python::python PROPERTY
+        INTERFACE_LINK_LIBRARIES ${PYTHON_LIBRARIES})
+
     caffe2_update_option(USE_NUMPY OFF)
     if(NUMPY_FOUND)
       caffe2_update_option(USE_NUMPY ON)
-      include_directories(SYSTEM ${NUMPY_INCLUDE_DIR})
+      add_library(numpy::numpy INTERFACE IMPORTED)
```

Nit, perhaps it would be a bit cleaner what this dependency is, if it were to be called `numpy_headers`
```suggestion
      add_library(numpy::numpy_headers INTERFACE IMPORTED)
```

---

IMO NumPy's interface being header-only is just an implementation detail and doesn't need to be advertised like that. Is that resonable?